cmd/scale.go

```diff
@@ -237,8 +239,31 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				vmsToDelete = append(vmsToDelete, indexToVM[i])
 			}
 
-			if orchestratorInfo.OrchestratorType == api.Kubernetes {
-				err = sc.drainNodes(vmsToDelete)
+			switch orchestratorInfo.OrchestratorType {
+			case api.Kubernetes:
+				kubeConfig, err := acsengine.GenerateKubeConfig(sc.containerService.Properties, sc.location)
+				if err != nil {
+					log.Errorf("failed to generate kube config: %v", err)
+					return err
```

Comment:
Logging an error and returning it seems redundant. What is the caller doing with it? Maybe just 
```go
return fmt.Errorf("failed to generate kube config: %v", err)
```

Reply:
This is a code move for k8s and I'm reluctant to refactor.  I'll ensure our newly added code is cleaner in this regard.  https://github.com/Azure/acs-engine/issues/2913 for the other code.

Reply:
Thanks for filing the backlog item ❤️